src/OpenSage.Game/Game.cs

```diff
@@ -102,14 +102,14 @@ public void LoadReplayFile(FileSystemEntry replayFileEntry)
         {
             var replayFile = ReplayFile.FromFileSystemEntry(replayFileEntry);
 
-            // TODO: This probably isn't right.
-            var mapFilenameParts = replayFile.Header.Metadata.MapFile.Split('/');
-            var mapFilename = $"Maps\\{mapFilenameParts[1]}\\{mapFilenameParts[1]}.map";
+            var mapFilename = replayFile.Header.Metadata.MapFile.Replace("userdata", _userDataFileSystem?.RootDirectory);
+            mapFilename = FileSystem.NormalizeFilePath(mapFilename);
+            var mapName = mapFilename.Substring(mapFilename.LastIndexOf(Path.DirectorySeparatorChar));
 
             var pSettings = ParseReplayMetaToPlayerSettings(replayFile.Header.Metadata.Slots);
 
             StartMultiPlayerGame(
-                mapFilename,
+                mapFilename + mapName + ".map",
```

Comment:
Isn't `mapFilename` already the entire path?

Reply:
No, mapFilename is something like "Maps/Homeland Alliance", whereas StartMultiPlayerGame expects the full path to the map file.